streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -363,6 +362,98 @@ public synchronized final TopologyBuilder addSource(String name, Deserializer ke
     }
 
     /**
+     * Add a global store to the topology. This store can be accessed by any of the Processors
+     * in the topology.
+     *
+     * @param store  the instance of {@link StateStore}
+     * @return this builder instance so methods can be chained together; never null
+     */
+    public synchronized TopologyBuilder addGlobalStore(final StateStore store) {
```

Comment:
I understand that argument. My point is more about API design -- we break an API pattern and thus reveal something to the user, that you can consider an implementation detail. The user does not care how ofter we do instantiate a store. If I write code for a store, and hand it in, I don't care about it -- I want to same API for regular and global stores. Why should I care about the number of instantiation as a user?

I personally dislike the whole handing in factories instead of stores from a user perspective completely -- even if I understand why it is necessary for Streams. (Flink for example has a very nice API for this and it's not a concern there.) As a user, I want to implement a store -- I don't want to bother with a store factory (I don't want to change this pattern because we have good reasons to enforce is, and one more wrapper for the store is an acceptable burden for the user IMHO). So we educate the user to implement factories and suddenly we change our mind and say -- "not for global store".